### DISCUSSION TITLE
node2vec example gives the same loss always

### DISCUSSION THREAD

**Discussion by MohamedEldafrawi2** [pyg-team/pytorch_geometric#5428] [General]
Hi, 
I was trying to run the node2vec example but I am always getting the same loss regardless how many epochs i have. 
Epoch: 01, Loss: 1.3863, Acc: 0.1300
Epoch: 02, Loss: 1.3863, Acc: 0.1300
Epoch: 03, Loss: 1.3863, Acc: 0.1300
Epoch: 04, Loss: 1.3863, Acc: 0.1300
Epoch: 05, Loss: 1.3863, Acc: 0.1300
Epoch: 06, Loss: 1.3863, Acc: 0.1300
Epoch: 07, Loss: 1.3863, Acc: 0.1300
Epoch: 08, Loss: 1.3863, Acc: 0.1300
Epoch: 09, Loss: 1.3863, Acc: 0.1300
Epoch: 10, Loss: 1.3863, Acc: 0.1300
Epoch: 11, Loss: 1.3863, Acc: 0.1300
Epoch: 12, Loss: 1.3863, Acc: 0.1300
Epoch: 13, Loss: 1.3863, Acc: 0.1300
Epoch: 14, Loss: 1.3863, Acc: 0.1300
Epoch: 15, Loss: 1.3863, Acc: 0.1300
Epoch: 16, Loss: 1.3863, Acc: 0.1300
Epoch: 17, Loss: 1.3863, Acc: 0.1300
Epoch: 18, Loss: 1.3863, Acc: 0.1300
Epoch: 19, Loss: 1.3863, Acc: 0.1300
Epoch: 20, Loss: 1.3863, Acc: 0.1300
Epoch: 21, Loss: 1.3863, Acc: 0.1300
Epoch: 22, Loss: 1.3863, Acc: 0.1300
Epoch: 23, Loss: 1.3863, Acc: 0.1300
Epoch: 24, Loss: 1.3863, Acc: 0.1300

Could you please tell me what is the reason behind that?

**Comment by EdisonLeeeee** [comment]
I guess there must be something wrong with your training process, e.g., the model parameters are not updated during training. Could you please show more details so that I can dig into it?

**Reply by MohamedEldafrawi2** [reply]
Thanks for your response. I didn't modify anything I am just trying to run this example https://github.com/pyg-team/pytorch_geometric/blob/master/examples/node2vec.py. 
I am using Pytorch 1.12 and Cuda 11.6. 

Thanks

**Reply by EdisonLeeeee** [reply]
That's weird. It works on my end. Which PyG version are you using?

**Reply by MohamedEldafrawi2** [reply]
2.1.0  windows machine

**Reply by MohamedEldafrawi2** [reply]
I was trying to debug it and I found out that the Embedding layer always returns zeros! inside (Loss) method >>Node2Vec class

**Reply by EdisonLeeeee** [reply]
Interesting. I can indeed reproduce this bug only on the Windows machine. Will figure it out and report back.

**Reply by MohamedEldafrawi2** [reply]
Thanks!

**Reply by EdisonLeeeee** [reply]
This is a bug caused by PyTorch Dataloader when `num_workers>0`, as also mentioned in https://github.com/pytorch/pytorch/issues/51344. To resolve this bug, just set `num_workers=0` in line 22, i.e.,
```python
 loader = model.loader(batch_size=128, shuffle=True, num_workers=0)
```

CC @rusty1s

**Reply by MohamedEldafrawi2** [reply]
Now it works! 
Thanks a lot.
